app/controllers/approval_requests_controller.rb

```diff
@@ -1,8 +1,68 @@
+# Normally, each vacation request has approval request assigned to a manager.
+# When approval request is accepted, it is deleted from DB.
+# When vacation request has no approval requests, it changes its status
+# from 'requested' to 'accepted'.
+# Approval request can be declined. In this case all the approval requests
+# are deleted from DB, and status of the associated vacation request
+# is set to 'declined'.
+# Only managers are authorized to operate on approval requests.
+require 'errors/conflict_error'
+
 class ApprovalRequestsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_approval_request, only: [:accept, :decline]
+  before_action :check_vacation_request_status, only: [:accept, :decline]
+
+  rescue_from ActiveRecord::RecordNotFound do
+    head status: :not_found
+  end
+
+  rescue_from Errors::ConflictError do
+    head status: :conflict
+  end
+
   def index
     vacation_requests_ids = current_user.approval_requests
                             .pluck(:vacation_request_id).uniq
     requests = VacationRequest.find vacation_requests_ids
     render json: requests
   end
+
+  def accept
+    authorize @approval_request
+    status = VacationRequest.statuses[:accepted]
+    change_vacation_request_status(status) if approval_request_count == 1
+    @approval_request.destroy
+    head status: :ok
+  end
+
+  def decline
+    # authorize @approval_request
```

Comment:
Aha. sounds good. But, personally I don't like this idea, because you add gem for this, when app is growing and a lot of libs make app hard. Second, writing test is always help other people understand you code faster. So, i'll chose writing tests. Gang @eleidan @rubycop @epmlys  does it make sense?